examples/horizontal_convection.jl

```diff
@@ -0,0 +1,347 @@
+# # Horizontal convection example
```

Comment:
Maybe a silly question but why is this called horizontal convection?  This is an `x-z` slice and there is cooling at the top that causes convection in the vertical.  It looks like this might be trying to simulate the effect of cooling by ice above.  It seems like vertical convection to me.

Reply:
Because the imposed temp difference is in the horizontal instead of vertical like Rayleigh Bernard.

Reply:
I didn’t came up with the name..

Reply:
Thanks.  In Rayleigh Bernard convection the heating is done at the bottom and is uniform in the horizontal direction, I agree.  In this problem, we have cooling at the top and it varies I the horizontal.  It seems like the temperature difference is both in the vertical and the horizontal.   I'm okay with this name but do wonder if we can find something more descriptive.

Reply:
There is both heating and cooling at the top.

Reply:
I’m not arguing that is the best or more descriptive name. But it’s what has prevailed over the years. For example: “Rayleigh Bernard convention” means nothing also, but nevertheless people know what that means.

Reply:
If this is what people tend to use in the literature, sorry for the distraction.  By all means we should keep the title as is.  I just haven't come across this problem so am new to the problem, but it is a ncie one.  By any name!